src/riak_kv_qry_worker.erl

```diff
@@ -168,6 +170,13 @@ make_key_conversion_fun(Table) ->
             Key
     end.
 
+%% Convert the sql select record to a version that is safe to pass around the
+%% cluster. Do not treat the result as a record in the local node, just as a
+%% tuple.
+convert_query_to_cluster_version(Query) ->
+    Version = riak_core_capability:get({riak_kv, sql_select_version}),
```

Comment:
If all nodes in the cluster do not define the capability, then the default value of 1 will be used.  Nodes that don't define it won't be calling it of course, and will just be using version 1 because that is all they know.